pyat/at/physics/fastring.py

```diff
@@ -100,3 +101,120 @@ def fast_ring(ring: Lattice, split_inds=[]) -> Tuple[Lattice, Lattice]:
                          split_inds=split_inds,
                          detuning_elem=detuning_elem)
     return fastringnorad, fastringrad
+
+
+def simple_ring(energy: float, circumference: float, harmonic_number: int,
+                Qx: float, Qy: float, Vrf: float, alpha: float,
+                beta_x: Optional[float]=1.0, beta_y: Optional[float]=1.0,
+                alpha_x: Optional[float]=0.0, alpha_y: Optional[float]=0.0,
+                Qpx: Optional[float]=0.0, Qpy: Optional[float]=0.0,
+                A1: Optional[float]=0.0, A2: Optional[float]=0.0,
+                A3: Optional[float]=0.0, emit_x: Optional[float]=0.0,
+                emit_y: Optional[float]=0.0, sigma_dp: Optional[float]=0.0,
+                tau_x: Optional[float]=0.0, tau_y: Optional[float]=0.0,
+                tau_z: Optional[float]=0.0, U0: Optional[float]=0.0
+                ):
+    """Generates a "simple ring" based on a given dictionary
+       of global parameters
+
+    A simple ring consists of:
+
+    * an RF cavity,
+    * a 6x6 linear transfer map,
+    * a detuning and chromaticity element,
+    * a simplified quantum diffusion element
+        which contains equilibrium emittance and radiation damping
+
+    Positional Arguments:
+        * energy [eV]
+        * circumference [m]
+        * harmonic_number
+        * Qx - horizontal tune
+        * Qy - vertical tune
+        * Vrf - RF Voltage set point [V]
+        * alpha (momentum compaction factor)
+
+    Optional Arguments:
+        * beta_x, beta_y [m]
+        * alpha_x, alpha_y
+        * Qpx, Qpy - linear chromaticities
+        * A1, A2, A3 - amplitude detuning coefficients
+        * emit_x, emit_y, sigma_dp - equilibrium values [m.rad, m.rad, -]
+        * tau_x, tau_y, tau_z - radiation damping times [turns]
+        * U0 - energy loss [eV] (positive number)
+
+    If the given emit_x,emit_y or sigma_dp is 0, then no equlibrium emittance is
+    applied in this plane.
+    If the given tau is 0, then no radiation damping is applied for this plane.
+
+
+    Returns:
+        ring (Lattice):    Simple ring
+    """
+
+    # compute rf frequency
+    frf = harmonic_number * clight / circumference
+
+    # compute slip factor
+    gamma = energy / e_mass
+    eta = alpha - 1/gamma**2
+
+    # compute the synchronous phase and the TimeLag
+    phi_s = numpy.arcsin(U0/Vrf)
+    TimeLag = clight * phi_s / (2 * numpy.pi * frf)
+    # generate rf cavity element
+    rfcav = RFCavity('RFC', 0, Vrf, frf, harmonic_number, energy,
```

Comment:
Have you thought about harmonic cavities?

Reply:
implemented array or scalar input for harmonic_number and Vrf

Reply:
The harmonic number is a lattice property, rather than a cavity property. Harmonic cavities should be defined by their harmonic with respect to the main frequency (3, 5, …). But that would make the input more complicated, so we could keep it like that.

When building the `Lattice` object, the constructor looks for the lowest frequency cavity and derives the lattice harmonic number from its harmonic number. So in the end, it will be correct.